internal/migrate/is_validated.go

```diff
@@ -42,45 +65,239 @@ var IsValidatedCommand = &command.Command{
 		Args:    cobra.MinimumNArgs(1),
 	},
 	Flags: &isValidatedflags,
-	RunS:  isValidated,
+	RunS:  isValidated(nil),
 }
 
-func isValidated(
+const (
+	repoOwner = "onflow"
+	repoName  = "cadence"
+	repoPath  = "migrations_data/raw"
+	repoRef   = "master"
+)
+
+func isValidated(repoService GitHubRepositoriesService) func(
 	args []string,
-	globalFlags command.GlobalFlags,
-	_ output.Logger,
+	_ command.GlobalFlags,
+	logger output.Logger,
 	flow flowkit.Services,
 	state *flowkit.State,
 ) (command.Result, error) {
-	err := checkNetwork(flow.Network())
+	return func(
+		args []string,
+		globalFlags command.GlobalFlags,
+		logger output.Logger,
+		flow flowkit.Services,
+		state *flowkit.State,
+	) (command.Result, error) {
+		err := checkNetwork(flow.Network())
+		if err != nil {
+			return nil, err
+		}
+
+		if repoService == nil {
+			repoService = github.NewClient(nil).Repositories
+		}
+
+		logger.StartProgress("Checking if contract has been validated")
+		defer logger.StopProgress()
+
+		contractName := args[0]
+		addr, err := getAddressByContractName(state, contractName, flow.Network())
+		if err != nil {
+			return nil, err
+		}
+
+		status, timestamp, err := getContractValidationStatus(
+			flow.Network(),
+			addr.HexWithPrefix(),
+			contractName,
+			state,
+			repoService,
+			logger,
+		)
+		if err != nil {
+			return nil, err
+		}
+
+		return validationResult{
+			Timestamp: *timestamp,
+			Status:    *status,
+		}, nil
+	}
+}
+
+func getContractValidationStatus(network config.Network, address string, contractName string, state *flowkit.State, repoService GitHubRepositoriesService, logger output.Logger) (*contractUpdateStatus, *time.Time, error) {
```

Comment:
Is it necessary to return a pointer here for `contractUpdateStatus`?

Reply:
Not necessary.  I can use an empty struct in place of everywhere I used nil, and it just felt more natural when doing the search inside of the for-loop (i.e. nil for object that may not exist), I don't really know what convention is here